app/controllers/api/v0/comments_controller.rb

```diff
@@ -10,7 +10,7 @@ def index
 
         @comments = article.comments.includes(:user).select(%i[id processed_html user_id ancestry]).arrange
 
-        set_surrogate_key_header "comments", edge_cache_keys(@comments)
+        set_surrogate_key_header article.record_key, "comments", edge_cache_keys(@comments)
```

Comment:
without the article's record key, the first hist of `/api/comments?a_id=ARTICLE_ID` becomes the response for ALL ids for 24 hours